### PR TITLE
Fixes Issue 56

### PR DESCRIPTION
**Pull request by sdellis**
Still having trouble with the Floor controller tests due to path conventions that the tests don't seem to be aware of, but features work as expected.

### CODE REVIEW

File: app/views/locations/libraries/show.html.erb [hunk 1]
```diff
@@ -8,6 +8,70 @@
...
+        <% end %>
+      </ul>
+      <% else %>
+        No floors specified.
+      <% end %>
+		</dd>
+	</dl>
+</p>
+
+<!--
```
Comment (kevinreiss) [inline app/views/locations/libraries/show.html.erb]: Does this commented out block need to stay for possible future use? Otherwise we should take it out.

**Comment by kevinreiss** [comment]
@sdellis Could you add a test for the validation flagged here, that will help raise the coveralls percentage, https://coveralls.io/builds/7092271/source?filename=app%2Fmodels%2Flocations%2Flibrary.rb. Could you also rebase this into a single commit? Thanks.

File: spec/controllers/locations/floors_controller_spec.rb [hunk 2]
```diff
@@ -51,71 +51,75 @@ module Locations
...
       end
 
     end
 
     describe "GET #show" do
       it "assigns the requested floor as @floor" do
         floor = Floor.create! valid_attributes
-        get :show, {:id => floor.to_param}, valid_session
-        expect(assigns(:floor)).to eq(floor)
+        #get :show, {:id => floor.to_param}, valid_session
```
Comment (kevinreiss) [inline spec/controllers/locations/floors_controller_spec.rb]: I think you can get all these tests to pass if you define the library_id param for the route. All the nested routes require two params, one for the floor, another for the library. Something like this should work:

```
get :show, {:library_id => floor.locations_library_id, :id => floor.to_param}, valid_session```

File: app/views/locations/libraries/show.html.erb (hunk 1 above)
Comment (sdellis) [inline app/views/locations/libraries/show.html.erb]: No, sorry, taking it out...

File: spec/controllers/locations/floors_controller_spec.rb (hunk 2 above)
Comment (sdellis) [inline spec/controllers/locations/floors_controller_spec.rb]: I can add a validation test there.  Assuming that goes in the Library test, right? Or the Floor test? 

As for getting the tests to pass, I tried that earlier, but it does not work.  The spec log states that the route is incorrect.  The app knows about the nested routes, but somehow the Floor controller tests do not.

```
Locations::FloorsController GET #index assigns all floors as @floors
     Failure/Error: get :index, {:locations_library_id => floor.locations_library_id, :id => floor.to_param}, valid_session ActionController::UrlGenerationError:
       No route matches {:action=>"index", :controller=>"locations/floors", :id=>"1", :locations_library_id=>"1"}
```

Comment (kevinreiss) [inline spec/controllers/locations/floors_controller_spec.rb]: When I pull this branch and change the test to read
```
describe "GET #show" do
      it "assigns the requested floor as @floor" do
        floor = Floor.create! valid_attributes
        get :show, {:library_id => floor.locations_library_id, :id => floor.to_param}, valid_session
        expect(assigns(:floor)).to eq(floor)
        #skip("Sort out Floor expected issues as nested resource.")
      end
    end
```
It passes.

Comment (sdellis) [inline spec/controllers/locations/floors_controller_spec.rb]: Ah, cool! Thanks! How did you know that the :library_id symbol doesn't need the "locations" namespace whereas the property does?

Comment (kevinreiss) [inline spec/controllers/locations/floors_controller_spec.rb]: I just looked at the symbol values used in the various routes output by ```rake routes```

File: spec/controllers/locations/floors_controller_spec.rb [hunk 3]
```diff
@@ -93,29 +92,34 @@ module Locations
           expect {
             post :create, {:floor => valid_attributes}, valid_session
```
Comment (kevinreiss) [inline spec/controllers/locations/floors_controller_spec.rb]: The create and update routes all require a ```:library_id``` param. 
```post :create, {:library_id => valid_attributes[:locations_library_id], :floor => valid_attributes}, valid_session``` allows the application accept the route. After that another error comes up about a ```floor_url``` method for the floor model.

**Comment by kevinreiss** [comment]
Could you add a test to address this coveralls flag? https://coveralls.io/builds/7125504/source?filename=app%2Fmodels%2Flocations%2Flibrary.rb

**Comment by sdellis** [comment]
Hmm, yeah, looking into it...

**Pull request merged**